Framework/CurveFitting/src/Functions/ProcessBackground.cpp

```diff
@@ -303,41 +305,38 @@ void ProcessBackground::deleteRegion() {
         "Lower boundary cannot be equal or larger than upper boundary.");
   }
 
-  // Copy original data and exclude region defined by m_lowerBound and
-  // m_upperBound
-  const MantidVec &dataX = m_dataWS->readX(0);
-  const MantidVec &dataY = m_dataWS->readY(0);
-  const MantidVec &dataE = m_dataWS->readE(0);
-
-  std::vector<double> vx, vy, ve;
+  const auto &dataX = m_dataWS->x(0);
+  const auto &dataY = m_dataWS->y(0);
+  const auto &dataE = m_dataWS->e(0);
 
-  for (size_t i = 0; i < dataY.size(); ++i) {
-    double xtmp = dataX[i];
-    if (xtmp < m_lowerBound || xtmp > m_upperBound) {
-      vx.push_back(dataX[i]);
-      vy.push_back(dataY[i]);
-      ve.push_back(dataE[i]);
+  // Find the dimensions of the region excluded by m_lowerBound and m_upperBound
+  std::vector<size_t> incIndexes;
+  for (size_t i = 0; i < dataY.size(); i++) {
+    if (dataX[i] < m_lowerBound || dataX[i] > m_upperBound) {
+      incIndexes.push_back(i);
     }
   }
+  size_t sizex, sizey;
+  sizex = sizey = incIndexes.size();
```

Comment:
I don't see what the problem is, its just a formality from the old code to make things slightly more clear.